hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieRealtimeTableCompactor.java

```diff
@@ -205,6 +199,10 @@ public HoodieCompactionWorkload generateCompactionWorkload(JavaSparkContext jsc,
     // compactions only
     HoodieCompactionWorkload workload = config.getCompactionStrategy().generateCompactionWorkload(config, operations,
         CompactionUtils.getAllPendingCompactionWorkloads(metaClient).stream().map(Pair::getValue).collect(toList()));
+    Preconditions.checkArgument(workload.getOperations().stream()
+            .filter(op -> fileIdsWithPendingCompactions.contains(op.getFileId())).count() == 0,
+        "FileId MUST NOT have multiple pending compactions. "
+            + "FileIdsWithPendingCompactions :" + fileIdsWithPendingCompactions + ", Selected workload :" + workload);
```

Comment:
Make a comment here that manual intervention is required.

Reply:
As this bad compaction plan is not set, I have added a message along the lines of "Please fix your strategy implementation."